driver-core/src/main/com/mongodb/connection/ServerDescription.java

```diff
@@ -773,85 +773,111 @@ public boolean equals(final Object o) {
 
         ServerDescription that = (ServerDescription) o;
 
-        if (maxDocumentSize != that.maxDocumentSize) {
+        if (shouldPublishChangeEvent(this, that)) {
             return false;
         }
-        if (ok != that.ok) {
+
+        if (maxDocumentSize != that.maxDocumentSize) {
             return false;
         }
-        if (!address.equals(that.address)) {
+        if (minWireVersion != that.minWireVersion) {
             return false;
         }
-        if (!arbiters.equals(that.arbiters)) {
+        if (maxWireVersion != that.maxWireVersion) {
             return false;
         }
-        if (canonicalAddress != null ? !canonicalAddress.equals(that.canonicalAddress) : that.canonicalAddress != null) {
+        if (lastWriteDate != null ? !lastWriteDate.equals(that.lastWriteDate) : that.lastWriteDate != null) {
             return false;
         }
-        if (!hosts.equals(that.hosts)) {
+        if (lastUpdateTimeNanos != that.lastUpdateTimeNanos) {
             return false;
         }
-        if (!passives.equals(that.passives)) {
+        if (logicalSessionTimeoutMinutes != null
+                    ? !logicalSessionTimeoutMinutes.equals(that.logicalSessionTimeoutMinutes)
+                    : that.logicalSessionTimeoutMinutes != null) {
             return false;
         }
-        if (primary != null ? !primary.equals(that.primary) : that.primary != null) {
-            return false;
+
+        return true;
+    }
+
+    /**
+     * Returns true if this instance can be published for SDAM change events.
+     *
+     * @param currentDescription the current server description
+     * @param previousDescription the previous server description
+     * @return true if this instance can be published for SDAM change events
+     */
+    public static boolean shouldPublishChangeEvent(final ServerDescription currentDescription,
```

Comment:
I'm still not quite happy with this, now I see it in its new location.  I'm wondering, does it need to be part of our public API?  I'm thinking that it shouldn't, and instead we should just put this somewhere less public, perhaps a new helper class in the same package as the classes that actually need it.

Reply:
I have put this method in its own helper class, `ServerDescriptionChangeEventHelper`.